app/src/main/java/com/sopt/now/LoginActivity.kt

```diff
@@ -2,57 +2,66 @@ package com.sopt.now
+
 
 class LoginActivity : AppCompatActivity() {
     private lateinit var binding: ActivityLoginBinding
+    private lateinit var viewModel: LoginViewModel
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityLoginBinding.inflate(layoutInflater)
-        fun showSnacker(view: View, message: String) {
-            Snackbar.make(
-                view,
-                message,
-                Snackbar.LENGTH_SHORT
-            ).show()
-        }
         setContentView(binding.root)
-        val userData = intent.getParcelableExtra<UserDataInput>(INTENT_USER_DATA)
-
-        val signId = userData?.getUserSignUpId()
-        val signPw = userData?.getUserSignUpPw()
-
-        binding.btnLogin.setOnClickListener {
-            val editid = binding.etId.text.toString()
-            val editpw = binding.pw2.text.toString()
-            if (editid == signId && editpw == signPw) {
-                showSnacker(binding.root, getString(R.string.log_in_success))
 
+        viewModel = ViewModelProvider(this).get(LoginViewModel::class.java)
+        initViews()
+        observeLoginViewModel()
 
-                val intent = Intent(this@LoginActivity, MainActivity::class.java).apply {
-                    putExtra(MainActivity.INTENT_USER_DATA, userData)
+    }
 
-                }
-                Log.d("IntentData", "User data: $userData")
-                startActivity(intent)
-                finish()
-            } else {
-                showSnacker(binding.root, getString(R.string.log_in_fail))
-            }
+    private fun initViews() {
+        binding.btnLogin.setOnClickListener {
+            val loginRequest = getLogInRequestDto()
+            viewModel.logIn(loginRequest)
         }
         binding.btnSignup.setOnClickListener {
             val intent = Intent(this@LoginActivity, SignUpActivity::class.java)
             startActivity(intent)
         }
     }
 
-    companion object {
-        const val INTENT_USER_DATA = "userData"
+    private fun observeLoginViewModel() {
+        viewModel.loginResult.observe(this, Observer { success ->
+            if (success) {
+                viewModel.userId.value?.let { userId ->
+                    val intent = Intent(this@LoginActivity, MainActivity::class.java)
+                    intent.putExtra("userId", userId)
+                    startActivity(intent)
+                    finish()
+                }
+            }
+        })
+
+        viewModel.errorMessage.observe(this, Observer { errorMessage ->
+            Toast.makeText(this@LoginActivity, errorMessage, Toast.LENGTH_SHORT).show()
+        })
+    }
+
+    private fun getLogInRequestDto(): RequestLogInDto {
+        val id = binding.etId.text.toString()
+        val password = binding.pw2.text.toString()
+
+        return RequestLogInDto(
+            authenticationId = id,
+            password = password
+        )
```

Comment:
```suggestion
        return RequestLogInDto(
            authenticationId = binding.etId.text.toString(),
            password = binding.pw2.text.toString()
        )
```
선호에 따라 다르겠지만 굳이 변수를 새로 할당해서 값을 넘겨주는 필요가 있을까 생각이 듭니다.